### DISCUSSION TITLE
How do I tell `craneLib.buildDepsOnly` that I am building only select Workspace Members?

### DISCUSSION THREAD

**Discussion by DrRuhe** [ipetkov/crane#204] [Q&A] [answered]
More Questions regarding [my attempts at packaging samply.](https://github.com/mstange/samply/pull/9)

Is it possible to build the dependencies only for a select crate? 

the motivating example:
samply itself does not require openssl, but the `tools/benchmarks` binary in the workspace does. For the purpose of packaging only samply, and not the benchmarks utility, openssl would not be necessary. Is there a way to only build the dependencies for a subset of the workspace members so that the openssl dependency is not required?

**Comment by dpc** [comment, marked as answer]
> Is it possible to build the dependencies only for a select crate?

Yes, you just have to make sure to pass `-p` to the build command in the `craneLib.buildDepsOnly`. Example from our codebase of a function that takes care of compiling only one package from the whole workspace

https://github.com/fedimint/fedimint/blob/c0651f88068b5818cb42a1d038d3b55d26b41e56/flake.nix#L421
